components/bl-carousel-component/src/index.js

```diff
@@ -0,0 +1,137 @@
+import { useState, useEffect, useRef } from 'react';
+import { NextButton, PrevButton, List, CarouselIndicators } from './subcomponents';
+import { getPrevImg, getNextImg } from './helpers';
+
+const { cn } = BackendlessUI.CSSUtils;
+
+export default function Carousel({ component, eventHandlers }) {
+  const {
+    display,
+    classList,
+    imagesData,
+    heightImage,
+    autoplayDelay,
+    animationDuration,
+    withControls,
+    withIndicators,
+    animationType
+  } = component;
+  const { onNextButton, onPrevButton, onMouseEnter, onMouseLeave } = eventHandlers;
+
+  const [data, setData] = useState(imagesData);
+  const [currentImg, setCurrentImg] = useState(0);
+  const [isNextAnimation, setIsNextAnimation] = useState(false);
+  const [isPrevAnimation, setIsPrevAnimation] = useState(false);
+  const [nextCurrentImage, setNextCurrentImage] = useState(0);
+  const [isAutoplay, setIsAutoplay] = useState(true);
+  const autoplayId = useRef();
+
+  const nextImg = getNextImg(currentImg, data.length);
+  const prevImg = getPrevImg(currentImg, data.length);
+
+  useEffect(() => {
+    if (autoplayDelay && isAutoplay) {
+      autoplayId.current = setTimeout(() => {
+        component.goToNextImage();
+      }, autoplayDelay);
+    } else if (!isAutoplay) {
+      clearTimeout(autoplayId.current);
+    }
+  }, [currentImg, autoplayDelay, isAutoplay]);
+
+  component.goToNextImage = () => {
+    setIsAutoplay(false);
```

Comment:
I'm not sure why we reassign autoplay after actions. Recreating a timeout is a good approach, but we don't need to change autoplay.